### PR TITLE
more-less: add remaining functionality

### PR DESCRIPTION
**Pull request by margaree**
This is mostly copy-pasted from the polymer 3 component with a few little modifications as necessary.
I added `d2l-dom.js` from `d2l-polymer-behaviors`. This could be potentially be renamed to something to do with ancestry.

I also fixed an issue with localization. I modified `proto.__localizationCache.requests[namespace] = true;` to be the result rather than `true` as this caused issues when multiple `more-less` components on the same page where the second component would go to get the resources, but the cache would say that they have already been fetched but they wouldn't be in `this.__resources` so the button text would be undefined.

### CODE REVIEW

File: components/helpers/d2l-dom.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+export function findComposedAncestor(node, predicate) {
```
Comment (dbatiste) [inline components/helpers/d2l-dom.js]: Ooooo... these are tidy now.  :)

**Comment by dbatiste** [comment]
I think `more-less` may be an extra good component for us to get started in the land of Lit...

I know this is mostly moved from the Polymer 3 version and it's ok to get the ball rolling, however rendering it Lit is done differently, and to get the benefits, I think we'll need to be careful when/how DOM and styles are changed and some refactoring may be needed.  :)

To make a long story short, Lit suggests that `render()` should be a pure function of data, it should not change the element's state, have side effects, only depend on element properties, returns the same result when called with the same property values.  Also any updates to the DOM or styles should be avoided outside of `render()`.

In the case of `more-less`, there are a wack of style updates that appear to happen outside the context of `render()`.  To apply styles within `render()` (based on properties as they suggest), `lit-html` provides [styleMap](https://lit-html.polymer-project.org/guide/styling-templates#stylemap) and [classMap](https://lit-html.polymer-project.org/guide/styling-templates#classmap) directives.  My current thinking is that we should try and update the `more-less` logic to follow this pattern (i.e. purely `UI = f(data)`).  I'm not intimately familiar with the `more-less` code, but it would probably require a bit of refactoring.  Might need to have a couple of private properties and update `render()` accordingly to use them.

Another interesting thing about `lit-element` is that it batches updates and applies them asynchronously.  This means that some of our `fastdom` calls may not be needed or desirable.  I am uncertain about this though.  We obviously would still like to avoid forced layout resulting from measures/mutations across components, but I am also worried an extra delay may be introduced by the batching/update process already at play with Lit elements.

If you haven't seen it, there is a nice blurb about this stuff. [Designing Performant Templates](https://lit-element.polymer-project.org/guide/templates#design-a-performant-template), and the [concept section](https://lit-html.polymer-project.org/guide/concepts#template-creation) in the `lit-html` gives some nice background on how the template is preprocessed, etc.

I have a feeling that this might come up a lot.  I can easily see folks updating DOM outside of `render()`.  It's an easy pit to fall in - I did at least initially with my `demo-transfigurator`... without even really noticing.  :(

**Comment by dlockhart** [comment]
That's a super good point Dave, thanks for sharing all that info. Seems like we could get these changes in (since it's the existing code), and then @margaree do you feel comfortable trying to attack the areas that are touching the DOM/style outside of render one by one? There seems like a lot to explore and learn here -- something we'll all need to get good at!

@dbatiste, do feel like you have enough information to be able to do some experiments around whether the usage of fastdom is still necessary with Lit?

**Comment by dbatiste** [comment]
Sounds good.  Yep - I can definitely explore the `fastdom` usage for Lit elements.

**Comment by margaree** [comment]
Sounds exciting! I can definitely get started on that.

**Comment by margaree** [comment]
Should I plan on merging this one and opening additional PR(s) for these changes? If so, any additional feedback? No rush, just want to see if I should be making changes here or separately.

**Comment by dbatiste** [comment]
I am good either way.

**Pull request merged**